### PR TITLE
Support for JSON & Binary Config Files

### PR DESCRIPTION
**Pull request by CleanCut**
Rebase and fixup of #2401

### CODE REVIEW

**Comment by CleanCut** [comment]
The (stable) CI failure seems to be entirely unrelated to this change. Is there breakage in master, currently?

The (nightly) CI failure seems to be a difference in `cargo fmt` behavior on nightly.

**Comment by Blisto91** [comment]
You the failure on ubuntu with rust beta?
Isn't it because you removed ``AssetStorage`` and ``IntoQuery`` but some code still depended on it?

File: amethyst_assets/tests/prefab_test.rs [hunk 1]
```diff
@@ -2,10 +2,10 @@ use std::time::{Duration, Instant};
 
 use amethyst_assets::{
     prefab::{register_component_type, Prefab},
-    AssetHandle, AssetStorage, DefaultLoader, Handle, LoadStatus, Loader, LoaderBundle,
+    AssetHandle, DefaultLoader, Handle, LoadStatus, Loader, LoaderBundle,
 };
 use amethyst_core::ecs::{
-    world::ComponentError, Dispatcher, DispatcherBuilder, Entity, IntoQuery, Resources, World,
```
Comment (Blisto91) [inline amethyst_assets/tests/prefab_test.rs:8]: ☝️

File: amethyst_assets/tests/prefab_test.rs [hunk 2]
```diff
@@ -2,10 +2,10 @@ use std::time::{Duration, Instant};
...
+    world::ComponentError, Dispatcher, DispatcherBuilder, Entity, Resources, World,
 };
```
Comment (Blisto91) [inline amethyst_assets/tests/prefab_test.rs:9]: ☝️

**Comment by ezpuzz** [comment]
example seems to work well, will merge after CI passes

**Review by ezpuzz — APPROVED** [review]

File: amethyst_assets/tests/prefab_test.rs (hunk 2 above)
Comment (CleanCut) [reply amethyst_assets/tests/prefab_test.rs]: I only manually touched files under `amethyst_config`. I must have missed that during either rebase or my fights with "rusty hook". 🤷🏼‍♂️

**Comment by CleanCut** [comment]
Thanks for taking this over the finish line. I'm confused about where the breaking changes _that I didn't make_ came from in my commit. I ran out of time on my lunch break and rushed a bit too fast. 😆

**Pull request merged**